plugins/X3DReader/X3DReader.py

```diff
@@ -0,0 +1,902 @@
+# Seva Alekseyev with National Institutes of Health, 2016
+
+from UM.Mesh.MeshReader import MeshReader
+from UM.Mesh.MeshBuilder import MeshBuilder
+from UM.Logger import Logger
+from UM.Math.Matrix import Matrix
+from UM.Math.Vector import Vector
+from UM.Scene.SceneNode import SceneNode
+from UM.Scene.GroupDecorator import GroupDecorator
+from UM.Job import Job
+from math import pi, sin, cos, sqrt
+import numpy
+
+
+EPSILON = 0.000001 # So very crude. :(
+
+try:
+    import xml.etree.cElementTree as ET
+except ImportError:
+    import xml.etree.ElementTree as ET
+    
+# TODO: preserve the structure of scenes that contain several objects
+# Use CADPart, for example, to distinguish between separate objects    
+    
+DEFAULT_SUBDIV = 16 # Default subdivision factor for spheres, cones, and cylinders
+
+class Shape:
+    def __init__(self, v, f, ib, n):
```

Comment:
A bit more documentation  or clearer names would be nice here. It doesn't do much now, but it could be that it will grow in size in the future. 
Personally i think that using self.verts = verts is better readable than self.verts = v